extensions/iam/decentralized-identity/identity-did-crypto/src/main/java/org/eclipse/dataspaceconnector/iam/did/crypto/credentials/VerifiableCredentialFactory.java

```diff
@@ -33,77 +31,45 @@
-            throw new CryptoException(e);
-        }
-    }
-
-    /**
-     * Creates a signed JWT {@link SignedJWT} that contains a set of claims and an issuer. Although all private key types are possible, in the context of Distributed Identity
-     * and ION using an Elliptic Curve key ({@code P-256}) is advisable.
-     *
-     * @param privateKey A Private Key represented as {@link ECKey}.
-     * @param claims a list of key-value-pairs that contain claims
-     * @param issuer the "owner" of the VC, in most cases this will be the DID ID. The VC will store this in the "iss" claim
-     * @param clock clock used to get current time
-     * @return a {@code SignedJWT} that is signed with the private key and contains all claims listed
-     */
-    public static SignedJWT create(ECKey privateKey, Map<String, String> claims, String issuer, Clock clock) {
-        return create(new EcPrivateKeyWrapper(privateKey), claims, issuer, clock);
-    }
+    // Subject claim value
+    public static final String VERIFIABLE_CREDENTIAL = "verifiable-credential";
 
     /**
      * Creates a signed JWT {@link SignedJWT} that contains a set of claims and an issuer. Although all private key types are possible, in the context of Distributed Identity and ION
      * using an Elliptic Curve key ({@code P-256}) is advisable.
      *
      * @param privateKey A Private Key represented as {@link PrivateKeyWrapper}.
-     * @param claims a list of key-value-pairs that contain claims
-     * @param issuer the "owner" of the VC, in most cases this will be the DID ID. The VC will store this in the "iss" claim
-     * @param clock clock used to get current time
+     * @param issuer     the "owner" of the VC, in most cases this will be the DID ID. The VC will store this in the "iss" claim
+     * @param audience   the audience of the token, e.g. the IDS Webhook address. The VC will store this in the "aud" claim
+     * @param clock      clock used to get current time
      * @return a {@code SignedJWT} that is signed with the private key and contains all claims listed
      */
-    public static SignedJWT create(PrivateKeyWrapper privateKey, Map<String, String> claims, String issuer, Clock clock) {
-        var claimssetBuilder = new JWTClaimsSet.Builder();
-
-        claims.forEach(claimssetBuilder::claim);
-        var claimsSet = claimssetBuilder.issuer(issuer)
-                .subject("verifiable-credential")
+    public static SignedJWT create(PrivateKeyWrapper privateKey, String issuer, String audience, Clock clock) {
+        var claimsSet = new JWTClaimsSet.Builder().issuer(issuer)
+                .issuer(issuer)
+                .subject(VERIFIABLE_CREDENTIAL)
+                .audience(audience)
                 .expirationTime(Date.from(clock.instant().plus(10, ChronoUnit.MINUTES)))
                 .jwtID(UUID.randomUUID().toString())
                 .build();
 
-        JWSSigner signer = privateKey.signer();
+        var signer = privateKey.signer();
```

Comment:
nit: Could (and imho should) stay `JWSSigner signer` as the type should not change and the method call does not reveal specific type information.

Reply:
IMHO this is not about consistency. Keywords should be used where they make sense.

However, if this is an EDC coding convention no change required here. 🤷‍♂️